velox/exec/fuzzer/PrestoQueryRunner.cpp

```diff
@@ -749,6 +798,60 @@ std::string PrestoQueryRunner::createTable(
   return tableDirectoryPath;
 }
 
+std::unordered_map<std::string, std::vector<velox::RowVectorPtr>>
+PrestoQueryRunner::getAllTablesAndNames(const core::PlanNodePtr& plan) {
+  std::unordered_map<std::string, std::vector<velox::RowVectorPtr>> result;
+  if (const auto valuesNode =
+          std::dynamic_pointer_cast<const core::ValuesNode>(plan)) {
+    result.insert({getTableName(valuesNode), valuesNode->values()});
+  } else {
+    for (const auto& source : plan->sources()) {
+      auto tablesAndNames = getAllTablesAndNames(source);
+      result.insert(tablesAndNames.begin(), tablesAndNames.end());
+    }
+  }
+  return result;
+}
+
+std::optional<std::vector<velox::RowVectorPtr>>
+PrestoQueryRunner::executeVector(const core::PlanNodePtr& plan) {
+  if (const auto sql = toSql(plan)) {
+    std::unordered_map<std::string, std::vector<velox::RowVectorPtr>> inputMap =
+        getAllTablesAndNames(plan);
+    std::unordered_map<std::string, std::vector<RowVectorPtr>>
+        inputMapWithNulls;
+    for (const auto& [tableName, input] : inputMap) {
+      auto inputType = asRowType(input[0]->type());
+      if (inputType->size() == 0) {
+        inputMapWithNulls.insert(
+            {tableName,
+             {makeNullRows(input, fmt::format("{}x", tableName), pool())}});
+      }
+    }
```

Comment:
Looks like you don't need a separate inputMapWithNulls. You can just modify inputMap?

Reply:
Yes! Done.